BuildingaMaster/AdventureGame
# MBUILD-28 Command Logger and Backtracking

## ~Note: On hold until all tests [MBUILD-141](https://csc-fdu.atlassian.net/browse/MBUILD-141) and [MBUILD-142](https://csc-fdu.atlassian.net/browse/MBUILD-142) are fixed.~

# Command Logger and Backtracking
A major UI and debugging upgrade.

### This PR handles:
* Receiving input from the terminal
* Arrow key commands
    * Up or down shows the last valid command inputted
    * Left or right moves the cursor

This PR also paves the way for the custom TerminalDisplay library used in the TicTacToe project.

### New requirements:
* NCurses Library (for macOS/Unix)


### How to test
* Typing in a command. 
* Pressing enter to submit a command
* Pressing up or down arrows
    * should beep (or play a sound) if there are no previous commands, or at the end of history.
* Left and right arrows move the cursors.
* YesNo Prompt cannot do up or down
* Google Tests still run correctly (with a few modifications)

[MBUILD-141]: https://csc-fdu.atlassian.net/browse/MBUILD-141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MBUILD-142]: https://csc-fdu.atlassian.net/browse/MBUILD-142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
